connector/github/github.go

```diff
@@ -380,43 +385,144 @@ func filterTeams(userTeams, configTeams []string) (teams []string) {
 	return
 }
 
-type user struct {
-	Name  string `json:"name"`
-	Login string `json:"login"`
-	ID    int    `json:"id"`
-	Email string `json:"email"`
-}
-
-// user queries the GitHub API for profile information using the provided client. The HTTP
-// client is expected to be constructed by the golang.org/x/oauth2 package, which inserts
-// a bearer token as part of the request.
-func (c *githubConnector) user(ctx context.Context, client *http.Client) (user, error) {
-	var u user
-	req, err := http.NewRequest("GET", c.apiURL+"/user", nil)
+// get creates a "GET `apiURL + path`" request with context, sends the request
+// using client, and decodes the resulting response body into v. The full HTTP
+// response object and any errors encountered are returned to facilitate post-
+// processing and error handling.
+func get(ctx context.Context, client *http.Client, apiURL, path string, v interface{}) (*http.Response, error) {
```

Comment:
Maybe do the pagination logic here and return something like `(nextPath string, err error)`?

And if the `nextPath != ""` there's another page.